distribution/lib/Standard/Visualization/0.0.0-dev/src/Text.enso

```diff
@@ -0,0 +1,51 @@
+from Standard.Base import all
+
+from Standard.Base.Data.Text.Extensions import slice_text
+
+## Return a sub-window of a string. The window is defined by line/chunk coordinates. The siz of
+   a chunk is defined by `chunk_width`. The output is formatted as a message that can be sent to
+   the IDE's lazy text visualisation.
+get_lazy_visualisation_text_window text pos size chunk_width =
+    get_text_chunk = get_item_from text chunk_width
+    pos_x = Math.max (pos.at 0) 0
+    pos_y = Math.max (pos.at 1) 0
+    size_x = size.at 0
+    size_y = size.at 1
+    x_range = pos_x.up_to (pos_x + size_x)
+    y_range = pos_y.up_to (pos_y + size_y)
+    coordinates = x_range.map (x -> y_range.map (y -> [x,y])) . flatten
+    chunks = coordinates.map (ix -> format_chunk ix (get_text_chunk ix))
+    data_string = format_chunks chunks
+    lines = text.lines.length
+    active_lines = y_range.map (line_ix -> text.lines.at line_ix)
+    max_line_length = (active_lines.map (line -> line.length)).fold 0 (l -> r -> Math.max l r)
+    make_grid_visualisation_response data_string lines max_line_length
+
+## Format a chunk of text and meta information for the lazy visualisation.
+make_grid_visualisation_response chunks lines max_line_length =
+    (Json.from_pairs [["chunks", chunks], ["lines", lines], ["max_line_length", max_line_length]]).to_text
```

Comment:
these names, like "chunks" or "lines" are hardcoded here. Can we instead just serialize an Enso struct instead of hardcoding them?